include/nbl/ext/CentralLimitBoxBlur/CBlurPerformer.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+#define _NBL_EXT_BLUR_C_BLUR_PERFORMER_INCLUDED_
+
+#include "nabla.h"
+
+namespace nbl
+{
+namespace ext
+{
+namespace CentralLimitBoxBlur
+{
+
+typedef uint32_t uint;
+struct alignas(16) uvec4
+{
+    uint x, y, z, w;
+};
+#include "nbl/builtin/glsl/ext/CentralLimitBoxBlur/parameters_struct.glsl"
+
+class CBlurPerformer final : public core::IReferenceCounted
+{
+public:
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t DEFAULT_WORKGROUP_SIZE = 256u;
+    _NBL_STATIC_INLINE_CONSTEXPR uint32_t PASSES_PER_AXIS = 3u;
+
+    typedef nbl_glsl_ext_Blur_Parameters_t Parameters_t;
```

Comment:
use this cool trick
```cpp
struct Parameters_t : impl::nbl_glsl_ext_Blur_Parameters_t 
{
};
```

then we can turn `buildParameters` from a static function to a constructor

Reply:
it could, but I'm a little hazy about a public class inheriting from a private one